File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.kt
/**
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */

class Solution {
    fun kthSmallest(root: TreeNode?, k: Int): Int {
        val res = mutableListOf<Int>()
        fun dfs(root: TreeNode?): TreeNode? {
            if (root== null)
                return null    
            
            dfs(root.left)
            res.add(root.`val`)
            dfs(root.right)

            return root
        }
        dfs(root)
        return res[k-1]
    }
}

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:   
    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool:
        res = []
        def serialize(root):
            if not root:
                res.append("None")
                return 
            
            res.append(str(root.val))
            serialize(root.left)
            serialize(root.right)

        serialize(root)
        root_st = ", ".join(res)
        res = []
        serialize(subRoot)
        subroot_st = ", ".join(res)

        return subroot_st in root_st